console/src/components/Services/Settings/Actions.ts

```diff
@@ -675,19 +732,70 @@ export const metadataReducer = (state = defaultState, action) => {
       return {
         ...state,
         allowedQueries: [
-          ...state.allowedQueries.filter(q => q.name !== action.data),
+          ...state.allowedQueries.filter((q: Query) => q.name !== action.data),
         ],
       };
     case UPDATE_ALLOWED_QUERY:
       return {
         ...state,
         allowedQueries: [
-          ...state.allowedQueries.map(q =>
-            q.name === action.data.queryName ? action.data.newQuery : q
-          ),
+          ...state.allowedQueries.map((q: Query) => {
+            const { queryName, newQuery } = action.data as UpdateAllowedQuery;
+            return q.name === queryName ? newQuery : q;
+          }),
         ],
       };
     default:
       return state;
   }
 };
+
+export type SettingsState = {
+  inconsistentObjects: object[];
+  ongoingRequest: boolean;
+  allowedQueries: Query[];
+};
+
+type ReloadConfig = {
+  shouldReloadMetadata?: boolean;
+  shouldReloadRemoteSchemas?: boolean;
+};
+
+type Query = { name: string; query: string };
+
+type UpdateAllowedQuery = { queryName: string; newQuery: Query };
+
+type MetadataReducerAction = {
+  type:
+    | typeof LOAD_INCONSISTENT_OBJECTS
+    | typeof LOADING_METADATA
+    | typeof LOAD_METADATA_ERROR
+    | typeof DROP_INCONSISTENT_METADATA
+    | typeof DROPPED_INCONSISTENT_METADATA
+    | typeof DROPPING_INCONSISTENT_METADATA_FAILED
+    | typeof LOAD_ALLOWED_QUERIES
+    | typeof ADD_ALLOWED_QUERIES
+    | typeof UPDATE_ALLOWED_QUERY
+    | typeof DELETE_ALLOWED_QUERY
+    | typeof DELETE_ALLOW_LIST;
+  data: Array<Query> | Query | string | Array<object> | UpdateAllowedQuery;
+};
+
+export type AppState = {
```

Comment:
Please add missing fields to `ReduxState` (`src/types.ts`) and remove `AppState`.